ModFrmHilD/GradedRing.m

```diff
@@ -611,3 +597,69 @@ intrinsic PrecomputeTraceForms(M::ModFrmHilDGRng, L::SeqEnum[RngOrdIdl])
   A := SetToSequence({ ii * aa : ii in Ideals(M), aa in L }); // Set of ideals
   HMFTracePrecomputation(M,A);
 end intrinsic;
+
+//////////////// Enumeration of Totally Positive Elements ////////////////
+
+intrinsic ElementsInABox(M::ModFrmHilDGRng, aa::RngOrdFracIdl,
+                         XLBound::Any, YLBound::Any, XUBound::Any, YUBound::Any) -> SeqEnum
+  {Enumerates all elements c in aa with XLBound <= c_1 <= XUBound and  YLBound <= c_2 <= YUBound}
```

Comment:
It looks like this function has to do with ordering by traces, is that correct?

Reply:
yes, it gets used in the trace formula code